File: src/test/kotlin/ch2/OverloadsTest.java
package ch2;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class OverloadsTest {

    @Test
    void checkOverloads() {
        Assertions.assertAll("overloads called from Java",
                () -> System.out.println(OverloadsKt.addProduct("Name", 5.0, "Desc")),
                () -> System.out.println(OverloadsKt.addProduct("Name", 5.0)),
                () -> System.out.println(OverloadsKt.addProduct("Name"))
        );
    }

}

File: src/test/kotlin/ch2/OverloadsTest.kt
package ch2

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertAll


@Test
fun `check all overloads`() {
    assertAll("Overloads called from Kotlin",
        { println(addProduct("Name", 5.0, "Desc")) },
        { println(addProduct("Name", 5.0)) },
        { println(addProduct("Name")) }
    )
}